account_due_list/account_move_line.py

```diff
@@ -0,0 +1,166 @@
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+
+from openerp.osv import fields, orm
+from openerp.tools.translate import _
+
+
+class account_move_line(orm.Model):
+
+    def _get_invoice(self, cr, uid, ids, field_name, arg, context=None):
+        invoice_pool = self.pool['account.invoice']
+        res = {}
+        for line in self.browse(cr, uid, ids, context=context):
+            inv_ids = invoice_pool.search(
+                cr,
+                uid,
+                [('move_id', '=', line.move_id.id)],
+                context=context
+            )
+            if len(inv_ids) > 1:
+                raise orm.except_orm(
+                    _('Error'),
+                    _('Inconsistent data: move %s has more than one invoice')
+                    % line.move_id.name
+                )
+            if inv_ids:
+                res[line.id] = inv_ids[0]
+            else:
+                res[line.id] = False
+        return res
+
+    def _get_day(self, cr, uid, ids, field_name, arg, context=None):
+        res = {}
+        for line in self.browse(cr, uid, ids, context=context):
+            if line.date_maturity:
+                res[line.id] = line.date_maturity
+            else:
+                res[line.id] = False
+        return res
+
+    def _get_move_lines(self, cr, uid, ids, context=None):
+        invoice_pool = self.pool['account.invoice']
+        res = []
+        for invoice in invoice_pool.browse(cr, uid, ids, context=context):
+            if invoice.move_id:
+                for line in invoice.move_id.line_id:
+                    if line.id not in res:
+                        res.append(line.id)
+        return res
+
+    _inherit = 'account.move.line'
+
+    _columns = {
+        'invoice_origin': fields.related(
+            'invoice',
+            'origin',
+            type='char',
+            string='Source Doc',
+            store=False
+        ),
+        'invoice_date': fields.related(
+            'invoice',
+            'date_invoice',
+            type='date',
+            string='Invoice Date',
+            store=False
+        ),
+        'partner_ref': fields.related(
+            'partner_id',
+            'ref',
+            type='char',
+            string='Partner Ref',
+            store=False
+        ),
+        'payment_term_id': fields.related(
+            'invoice',
+            'payment_term',
+            type='many2one',
+            string='Payment Term',
+            store=False,
+            relation="account.payment.term"
+        ),
+        'stored_invoice_id': fields.function(
```

Comment:
Because standard ```invoice``` field can't be used for grouping in this way: https://github.com/binovo/account-payment/blob/8.0-account_due_list/account_due_list/payment_view.xml#L54

At least in 7.0. Does 8.0 allow to group by not stored fields?